### PR TITLE
Update TagBot GHA; add CompatHelper and CI GHAs

### LINKED ISSUE #10: Add CI and CompatHelper GHAs

### PR DESCRIPTION
**Pull request by PerezHz**
Fixes #10. Also updates TagBot GHA.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
+      - uses: julia-actions/cache@v1
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v3
+        with:
+          files: lcov.info
+      - uses: julia-actions/julia-uploadcoveralls@v1
```
Comment (lbenet) [inline .github/workflows/CI.yml:41]: The errors in the tests are related to the upload process here

**Comment by lbenet** [comment]
Though not related to the main subject of this PR, few packages used (`Healpix`, `IntervalArithmetic`, `StatsBase` and `TaylorIntegration`) could (should?) be upgraded...

**Comment by PerezHz** [comment]
> Though not related to the main subject of this PR, few packages used (Healpix, IntervalArithmetic, StatsBase and TaylorIntegration) could (should?) be upgraded...

Just noticed PlanetaryEphemeris.jl v0.5.1 has not yet been released due to https://github.com/JuliaRegistries/General/pull/84252 not being merged yet; this causes in turn that NEOs.jl environment cannot be resolved

**Comment by lbenet** [comment]
> Just noticed PlanetaryEphemeris.jl v0.5.1 has not yet been released due to [JuliaRegistries/General#84252](https://github.com/JuliaRegistries/General/pull/84252) not being merged yet; this causes in turn that NEOs.jl environment cannot be resolved

There is a test failing in [JuliaRegistries/General#84252](https://github.com/JuliaRegistries/General/pull/84252), which may delay merging it.

File: Project.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name = "NEOs"
 uuid = "b41c07a2-2abb-11e9-070a-c3c1b239e7df"
 authors = ["Jorge A. Pérez Hernández", "Luis Benet"]
-version = "0.3.0"
+version = "0.3.1"
```
Comment (lbenet) [inline Project.toml:4]: I suggest to leave this as "0.3"; in includes all 0.3.x versions.

**Comment by PerezHz** [comment]
> There is a test failing in JuliaRegistries/General#84252, which may delay merging it.

I tried re-triggering JuliaRegistrator, but autoMerge failed again due to the process taking too long (again, probably due to precompilation during package loading)

File: .github/workflows/CI.yml (hunk 1 above)
Comment (PerezHz) [reply .github/workflows/CI.yml:41]: The errors caused by the coverage actions seem to be resolved now 👍

File: Project.toml (hunk 2 above)
Comment (PerezHz) [reply Project.toml]: Done.

**Comment by PerezHz** [comment]
> Though not related to the main subject of this PR, few packages used (Healpix, IntervalArithmetic, StatsBase and TaylorIntegration) could (should?) be upgraded...

I've now upgraded the dependencies compatibility requirements, including the update to PlanetaryEphemeris.jl v0.6.0 which has just been released.

> I suggest to leave this as "0.3"; in includes all 0.3.x versions.

I also update this, thanks!

**Comment by PerezHz** [comment]
We've got the green lights, so I'll go ahead and merge.

**Comment by PerezHz** [comment]
In the end, JuliaRegistrator [returned an error](https://github.com/PerezHz/NEOs.jl/commit/7387ccc5b86176c68ad9741f08fb33624bdf9009#commitcomment-115353106) regarding the NEOs.jl version stated as "0.3" in the Project.toml, which it interpreted as "0.3.0", so I pushed another commit onto main with the version "0.3.1".

**Pull request merged**